src/components/PostInfo/PostInfo.tsx

```diff
@@ -1,5 +1,45 @@
-import React from 'react';
+import { UserInfo } from '../UserInfo';
+import commentsFromServer from '../../api/comments';
+import usersFromServer from '../../api/users';
+import { Post } from '../../types/post';
+import { CommentList } from '../CommentList';
 
-export const PostInfo: React.FC = () => (
-  <>Put the post here</>
-);
+type Props = {
+  post: Post,
+};
+
+export const PostInfo: React.FC<Props> = ({ post }) => {
+  const {
+    userId,
+    title,
+    body,
+    id,
+  } = post;
+
+  const user = usersFromServer.find(curUser => curUser.id === userId);
+  const comments = commentsFromServer
+    .filter(curComment => curComment.postId === id);
+
+  return (
+    <div className="PostInfo">
+      <div className="PostInfo__header">
+        <h3 className="PostInfo__title">{title}</h3>
+
+        <p>
+          {' Posted by  '}
+
+          {user && <UserInfo dataType="post" user={user} />}
```

Comment:
interesting, why did you need dataType here? it wasn't in original file

Reply:
now I saw, okay)

Reply:
excuse me) i have to ask about i missed it maybe or cant understand what you about ((((

Reply:
> interesting, why did you need dataType here? it wasn't in original file

excuse me) i have to ask about i missed it maybe or cant understand what you about ((((

Reply:
we use UserInfo each post 2 times for each post with comments
once for author of the post (we show name)
and once for each comment (we show the email)

So i maid a component, which can do both. dataType is used to show which type we need in each iteration

Reply:
thanks, I'll try ro reread